keps/sig-network/2030-endpointslice-subsetting/README.md

```diff
@@ -0,0 +1,207 @@
+- Minimal duplication of data.
+- Room for future enhancements, for example weighted endpoints or Slices.
+
+### Non-Goals
+
+- Define how subsetting should be used.
+- Design the controller responsible for subsetting endpoints.
+- An API for telling controllers how a service should be subsetted.
+
+Many of these are being tackled by the follow up [KEP
+#2004](https://github.com/kubernetes/enhancements/issues/2004).
+
+## Proposal
+
+Two new topology based labels will be introduced for EndpointSlices to support
+subsetting:
+
+```
+endpointslice.kubernetes.io/for-zone
+endpointslice.kubernetes.io/for-region
+```
+
+In the future this pattern may be expanded to include other concepts or
+topologies. A simple pattern like this will allow EndpointSlices to be delivered
+to consumers in a specific zone or region.
+
+
+### Risks and Mitigations
+
+This approach does not allow a single EndpointSlice to target multiple zones or
+regions. Any approach that enabled that would be significantly more complicated.
+The initial proposal in [KEP
+#2004](https://github.com/kubernetes/enhancements/issues/2004) suggests that
+this won't be an issue. 
+
+## Design Details
+
+### API Consumption
+
+EndpointSlice API consumers can use the following selectors to consume
+EndpointSlices:
+
+#### Select EndpointSlices Without Labels for Backwards Compatibility
+```
+matchExpressions:
+  - {key: endpointslice.kubernetes.io/for-zone, operator: DoesNotExist}
+  - {key: endpointslice.kubernetes.io/for-region, operator: DoesNotExist}
+```
+
+#### Select EndpointSlices With Matching Zone
+```
+matchLabels:
+  endpointslice.kubernetes.io/for-zone: example-zone
+```
+
+#### Select EndpointSlices With Matching Region
+```
+matchLabels:
+  endpointslice.kubernetes.io/for-region: example-region
+```
+
+#### Kube-Proxy
+When the `EndpointSliceSubsetting` feature gate is set to true on `Kube-Proxy`,
+it will use these selectors to filter EndpointSlices.
+
+### Test Plan
+This KEP is quite small in scope. The only new functionality being added will be
+an adjustment to the EndpointSlices kube-proxy consumes when a feature gate is
+enabled. We will need to add more test coverage for when this feature is enabled
+or disabled.
+
+#### Unit Tests
+* Ensure kube-proxy will continue to consume all EndpointSlices when this
+  feature is disabled.
+* Ensure EndpointSlices delivered to a specific zone will be consumed by
+  kube-proxy running in the same zone when this feature is enabled.
+* Ensure EndpointSlices delivered to a specific zone will not be consumed by
+  kube-proxy running in a different zone when this feature is enabled.
+* Ensure EndpointSlices delivered to a specific region will be consumed by
+  kube-proxy running in the same region when this feature is enabled.
+* Ensure EndpointSlices delivered to a specific region will not be consumed by
+  kube-proxy running in a different region when this feature is enabled.
+
+### Graduation Criteria
+
```

Comment:
Could I get a line on the alpha deliverable?

Reply:
Added, thanks!